tools/nnyeah/nnyeah/MethodTransformations.cs

```diff
@@ -0,0 +1,832 @@
+			allTransforms.Add (new Transformation ("System.Boolean System.nuint::TryParse(System.String,System.nuint&)",
+				Instruction.Create (OpCodes.Call, mref)));
+
+			mref = new MethodReference ("TryParse", module.TypeSystem.Boolean, module.TypeSystem.UIntPtr);
+			mref.Parameters.Add (stringParam);
+			mref.Parameters.Add (globNumberStylesParam);
+			mref.Parameters.Add (formatProviderParam);
+			mref.Parameters.Add (nintRefParam);
+			mref = module.ImportReference (mref);
+			allTransforms.Add (new Transformation ("System.Boolean System.nuint::TryParse(System.String,System.Globalization.NumberStyles,System.IFormatProvider,System.nuint&)",
+				Instruction.Create (OpCodes.Call, mref)));
+
+			decimalTypeReference = new TypeReference ("System", "Decimal", module, module.TypeSystem.CoreLibrary);
+			mref = new MethodReference ("op_Implicit", decimalTypeReference, decimalTypeReference);
+			mref.Parameters.Add (new ParameterDefinition (module.TypeSystem.UInt64));
+			mref = module.ImportReference (mref);
+			allTransforms.Add (new Transformation ("System.Decimal System.nuint::op_Implicit(System.nuint)",
+				TransformationAction.Replace,
+				new List<Instruction> () {
+					Instruction.Create (OpCodes.Conv_U8),
+					Instruction.Create (OpCodes.Call, mref) }));
+
+			mref = new MethodReference ("op_Explicit", module.TypeSystem.UInt64, decimalTypeReference);
+			mref.Parameters.Add (new ParameterDefinition (decimalTypeReference));
+			mref = module.ImportReference (mref);
+			allTransforms.Add (new Transformation ("System.nuint System.nuint::op_Explicit(System.Decimal)",
+				TransformationAction.Replace,
+				new List<Instruction> () {
+					Instruction.Create (OpCodes.Call, mref),
+					Instruction.Create (OpCodes.Conv_U) }));
+
+			nfloatTypeRef = new TypeReference ("System.Runtime.InteropServices", "NFloat", module, module.TypeSystem.CoreLibrary);
+			mref = new MethodReference (".ctor", module.TypeSystem.Void, nfloatTypeRef);
+			mref.Parameters.Add (new ParameterDefinition (module.TypeSystem.Double));
+			mref = module.ImportReference (mref);
+			allTransforms.Add (new Transformation ("System.nfloat System.nuint::op_Implicit(System.nuint)",
+				TransformationAction.Replace,
+				new List<Instruction> () {
+					Instruction.Create (OpCodes.Conv_R8),
+					Instruction.Create (OpCodes.Newobj, mref) }));
+
+			var nuintVar = new VariableDefinition (module.TypeSystem.UInt64);
+			mref = new MethodReference ("GetTypeCode", typeCodeRef, module.TypeSystem.UInt64);
+			mref = module.ImportReference (mref);
+			allTransforms.Add (new Transformation ("System.TypeCode System.nuint::GetTypeCode()",
+				TransformationAction.Replace,
+				new List<Instruction> () {
+					Instruction.Create (OpCodes.Ldobj, module.TypeSystem.UIntPtr),
+					Instruction.Create (OpCodes.Conv_U8),
+					Instruction.Create (OpCodes.Stloc, nuintVar),
+					Instruction.Create (OpCodes.Ldloca, nuintVar),
+					Instruction.Create (OpCodes.Call, mref)
+				}));
```

Comment:
There seems to be a lot of code repetition in all the above calls, for example:
```csharp
			mref = new MethodReference ("CompareTo", module.TypeSystem.Int32, module.TypeSystem.UIntPtr);
			mref.Parameters.Add (new ParameterDefinition (module.TypeSystem.UIntPtr));
			parm.CustomAttributes.Add (attrBuilder (singleBool));
			allTransforms.Add (new Transformation ("System.Int32 System.nuint::CompareTo(System.nuint)",
				Instruction.Create (OpCodes.Call, mref)));

			mref = new MethodReference ("CompareTo", module.TypeSystem.Int32, module.TypeSystem.UIntPtr);
			mref.Parameters.Add (new ParameterDefinition (module.TypeSystem.Object));
			allTransforms.Add (new Transformation ("System.Int32 System.nuint::CompareTo(System.Object)",
				Instruction.Create (OpCodes.Call, mref)));

			mref = new MethodReference ("Equals", module.TypeSystem.Boolean, module.TypeSystem.UIntPtr);
			mref.Parameters.Add (new ParameterDefinition (module.TypeSystem.Object));
			allTransforms.Add (new Transformation ("System.Boolean System.nuint::Equals(System.Object)",
				Instruction.Create (OpCodes.Call, mref)));
```

They are very very similar, is there no way we could generalise some of the repetitive code and simply use a loop? I do see there are diff between some of them, is it possible to create a structure that defines the basic stuff and then loop? We are only populating a config to be used later, am I mistaken?

Reply:
Yes, it could be generalized, but I felt that was likely to cause more errors than in would prevent and make maintaining it more challenging.

Reply:
How about instead of trying to generalize (which I could see being a total pain) we make little builder functions to reduce the duplication.
```
ParameterDefinition ToParam (Type t) => new ParameterDefinition (module.TypeSystem.Object);
```
and such?